subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/swift/SwiftCompiler.java

```diff
@@ -138,11 +161,95 @@ public void execute(BuildOperationQueue<CommandLineToolInvocation> buildQueue) {
 
                 CommandLineToolInvocation perFileInvocation =
                     newInvocation("compiling swift file(s)", objectDir, Iterables.concat(genericArgs, outputArgs, importRootArgs), spec.getOperationLogger());
+                perFileInvocation.getEnvironment().put("TMPDIR", spec.getTempDir().getAbsolutePath());
                 buildQueue.add(perFileInvocation);
             }
         };
     }
 
+    /*
+version: "Swift version 4.0.3 (swift-4.0.3-RELEASE)"
+options: "7890c730e32273cd2686f36d1bd976c0"
+build_time: [1517422583, 339630833]
+inputs:
+  "fully-qualified-path/src/test/swift/BarTestSuite.swift": [9223372036, 854775807]
+  "fully-qualified-path/src/test/swift/main.swift": [1517422583, 0]
+  "fully-qualified-path/src/test/swift/FooTestSuite.swift": [1517422583, 0]
+     */
+    //CHECKSTYLE:OFF
+    public static class SwiftDeps {
+        private String version;
+        private String options;
+        private Integer[] build_time;
+        private Map<String, Integer[]> inputs;
+
+        public String getVersion() {
+            return version;
+        }
+
+        public void setVersion(String version) {
+            this.version = version;
+        }
+
+        public String getOptions() {
+            return options;
+        }
+
+        public void setOptions(String options) {
+            this.options = options;
+        }
+
+        public Integer[] getBuild_time() {
+            return build_time;
+        }
+
+        public void setBuild_time(Integer[] build_time) {
+            this.build_time = build_time;
+        }
+
+        public Map<String, Integer[]> getInputs() {
+            return inputs;
+        }
+
+        public void setInputs(Map<String, Integer[]> inputs) {
+            this.inputs = inputs;
+        }
+    }
+    //CHECKSTYLE:ON
+
+    private boolean adjustSwiftDepsForIncrementalCompile(File moduleSwiftDeps, Collection<File> changedSources) {
```

Comment:
I would feel better if we had an explicit unit test of this functionality that validated that we only change what we expect to change.

Reply:
done